event.go

```diff
@@ -537,16 +537,10 @@ func (e *Event) Redact() *Event {
 		// This is unreachable for events created with EventBuilder.Build or NewEventFromUntrustedJSON
 		panic(fmt.Errorf("gomatrixserverlib: invalid event %v", err))
 	}
-	result := Event{
-		redacted:    true,
-		roomVersion: e.roomVersion,
-		eventJSON:   eventJSON,
```

Comment:
We don't seem to set `eventJSON` to the redacted version?

Reply:
Yep, it's done by `populateFieldsFromJSON`.